activerecord/test/cases/associations/belongs_to_associations_test.rb

```diff
@@ -63,7 +63,7 @@ def test_belongs_to_does_not_use_order_by
     ActiveRecord::SQLCounter.clear_log
     Client.find(3).firm
   ensure
-    assert ActiveRecord::SQLCounter.log_all.all? { |sql| /order by/i !~ sql }, "ORDER BY was used in the query"
+    assert_no_match(/order by/i, ActiveRecord::SQLCounter.log_all.last, "ORDER BY was used in the query")
```

Comment:
How about just using `SQLCounter.log` instead of `SQLCounter.log_all`?

Reply:
And also,  looks like `test_has_one_does_not_use_order_by` also have the same problem.

https://github.com/rails/rails/blob/41ffddbc8b7faec66a26bd48dfd36a9def6cc23f/activerecord/test/cases/associations/has_one_associations_test.rb#L36-L42